agent/configuration-build-derive/src/derive.rs

```diff
@@ -0,0 +1,433 @@
+                        self
+                    }
+
+                    pub fn retries<S1>(&mut self, retries: u32) -> &mut Self {
+                        self.retries = Some(retries);
+                        self
+                    }
+
+                    pub fn set_retries(&mut self, retries: Option<u32>) -> &mut Self {
+                        self.retries = retries;
+                        self
+                    }
+
+                    pub fn keep_running<S1>(&mut self, keep_running: bool) -> &mut Self {
+                        self.keep_running = Some(keep_running);
+                        self
+                    }
+
+                    pub fn set_keep_running(&mut self, keep_running: Option<bool>) -> &mut Self {
+                        self.keep_running = keep_running;
+                        self
+                    }
+
+                    pub fn capabilities<S1>(&mut self, capabilities: S1) -> &mut Self
+                    where
+                    S1: Into<String> {
+                        self.capabilities.push(capabilities.into());
+                        self
+                    }
+
+                    pub fn set_capabilities(&mut self, capabilities: Option<Vec<String>>) -> &mut Self {
+                        self.capabilities = capabilities.unwrap_or_default();
+                        self
+                    }
+
+                    pub fn build<S1>(&self, name: S1) -> Result<model::Test, String>
+                    where
+                    S1: Into<String>,
+                    {
+                        let configuration =  match serde_json::to_value(self) {
+                            Ok(serde_json::Value::Object(map)) => map,
+                            Err(error) => return Err(format!("Unable to serialize config: {}", error)),
+                            _ => return Err("Configuration must be a map".to_string()),
+                        };
+
+                        Ok(model::create_test_crd(name, Some(&self.labels), model::TestSpec {
+                                resources: self.resources.clone(),
+                                depends_on: Some(self.depends_on.clone()),
+                                retries: Some(self.retries.as_ref().cloned().unwrap_or(5)),
+                                agent: model::Agent {
+                                    name: "agent".to_string(),
+                                    image: self.image.as_ref().cloned().ok_or_else(|| "Image is required to build a test".to_string())?,
+                                    pull_secret: self.image_pull_secret.as_ref().cloned(),
+                                    keep_running: self.keep_running.as_ref().cloned().unwrap_or(true),
+                                    timeout: None,
+                                    configuration: Some(configuration),
+                                    secrets: Some(self.secrets.clone()),
+                                    capabilities: Some(self.capabilities.clone()),
```

Comment:
I think that makes sense. We just may miss out on some updated agent features.

Reply:
I actually think it's better to have a compilation error so that I understand that there are extra steps involved with adding a field to `Agent`.

In otherwords, if I add `foo: String` to the `Agent` struct, then someone later would be very confused as to why the generated builder doesn't allow you to set `foo`. To avoid that, I would rather have the compiler fail during the derive macro so that I am forced to add it to this builder.